client/signup/steps/site-type/form.jsx

```diff
@@ -44,29 +52,88 @@ class SiteTypeForm extends Component {
 		this.props.submitForm( type );
 	};
 
-	renderSegmentOptions() {
-		return getAllSiteTypes().map( siteTypeProperties => (
-			<Card
-				className="site-type__option"
-				key={ siteTypeProperties.id }
-				displayAsLink
-				data-e2e-title={ siteTypeProperties.slug }
-				tagName="button"
-				onClick={ this.handleSubmit.bind( this, siteTypeProperties.slug ) }
-			>
-				<strong className="site-type__option-label">{ siteTypeProperties.label }</strong>
-				<span className="site-type__option-description">{ siteTypeProperties.description }</span>
+	handleSubmitOther = () => {
+		if ( ! this.state.otherValue || this.state.hasOtherReasonFocus ) {
+			return;
+		}
+		this.handleSubmit( 'other—' + this.state.otherValue );
+	};
+
+	setOtherReasonFocus = focus => () => {
+		this.setState( { hasOtherReasonFocus: focus } );
+	};
+
+	renderBasicCard = () => {
+		return (
+			<Card className="site-type__wrapper">
+				{ getAllSiteTypes().map( siteTypeProperties => (
+					<Card
+						className="site-type__option"
+						key={ siteTypeProperties.id }
+						displayAsLink
+						data-e2e-title={ siteTypeProperties.slug }
+						onClick={ this.handleSubmit.bind( this, siteTypeProperties.slug ) }
+					>
+						<strong className="site-type__option-label">{ siteTypeProperties.label }</strong>
+					</Card>
+				) ) }
 			</Card>
-		) );
-	}
+		);
+	};
+
+	renderOtherInfo = () => {
+		const { translate } = this.props;
+
+		return (
+			<div className="site-type__other-option">
+				<p className="site-type__other-label">{ translate( 'Or type your own' ) }</p>
+
+				<div className="site-type__other-form">
+					<FormTextInput
```

Comment:
And another question: should we be limiting the number of characters with a `maxlength` attribute here? Would be nice for the user so they know they're limited (although it is quite an edge case).

Reply:
I do not think it is necessary.

Reply:
Well, why not? We're limiting in the backend, so it only makes sense to limit in the frontend too, for consistency at least.

Also, I still think we could harness `FormTextInputWithAction` and remove some code.

Reply:
I do no think it is necessary atm since the design speaks for itself. If upon analyzing the results we see that the input is too lengthy, I'd add it.

Reply:
I don't see how a `maxlength` could hurt, but if you have so hard feelings to not add it, that's fine.

What are your thoughts about using `FormTextInputWithAction`?